File: Assignment/Chapter 1/Student Grade Calculator/Kotlin Version/app/src/main/java/com/romualdsigning/studentgradecalc/domain/model/ExportFormat.kt
package com.romualdsigning.studentgradecalc.domain.model

enum class ExportFormat(
    val label: String,
    val extension: String,
    val mimeType: String,
) {
    EXCEL(
        label = "Excel",
        extension = "xlsx",
        mimeType = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
    ),
    CSV(
        label = "CSV",
        extension = "csv",
        mimeType = "text/csv",
    ),
    JSON(
        label = "JSON",
        extension = "json",
        mimeType = "application/json",
    ),
    PDF(
        label = "PDF",
        extension = "pdf",
        mimeType = "application/pdf",
    ),
    WORD(
        label = "Word (RTF)",
        extension = "rtf",
        mimeType = "application/rtf",
    );

    fun suggestedFileName(stem: String = "student_grade_report"): String = "$stem.$extension"
}

File: Assignment/Chapter 1/Student Grade Calculator/Dart Version/lib/core/models/export_format.dart
enum ExportFormat {
  excel,
  csv,
  json,
  pdf,
  word,
}

extension ExportFormatX on ExportFormat {
  String get label => switch (this) {
    ExportFormat.excel => 'Excel',
    ExportFormat.csv => 'CSV',
    ExportFormat.json => 'JSON',
    ExportFormat.pdf => 'PDF',
    ExportFormat.word => 'Word (RTF)',
  };

  String get extension => switch (this) {
    ExportFormat.excel => 'xlsx',
    ExportFormat.csv => 'csv',
    ExportFormat.json => 'json',
    ExportFormat.pdf => 'pdf',
    ExportFormat.word => 'rtf',
  };

  String get mimeType => switch (this) {
    ExportFormat.excel =>
      'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    ExportFormat.csv => 'text/csv',
    ExportFormat.json => 'application/json',
    ExportFormat.pdf => 'application/pdf',
    ExportFormat.word => 'application/rtf',
  };

  String suggestedFileName([String stem = 'student_grade_report']) =>
      '$stem.$extension';
}
